src/Microsoft.DotNet.Build.Tasks/Targets/post-process-assembly.targets

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <IntermediateAssemblyPath>$(IntermediateOutputPath)$(TargetName)$(TargetExt)</IntermediateAssemblyPath>
+    <PostProcessAssemblyMarker>$(IntermediateAssemblyPath).postprocessed</PostProcessAssemblyMarker>
+  </PropertyGroup>
+
+  <Target Name="PostProcessAssembly"
+    AfterTargets="CoreCompile"
+    Inputs="$(IntermediateAssemblyPath)"
+    Outputs="$(PostProcessAssemblyMarker)">
+
+    <CallTarget Targets="@(PostProcessAssemblySteps)" />
```

Comment:
I see what you mean. It's a side effect of these steps happening in-place and the output being a marker file instead of a real output.  If you didn't do these in-place that might fix the problem.

One problem I see with this now is that incremental behavior of post process assembly only depends on the intermediateassemblypath, but if the seed or contract changes you also need to rerun it. Another thing I am not sure about is the behavior of CallTarget when one target fails, will you continue to execute and touch the marker?